### PR TITLE
ECCW-38: Added openid config to be populated via env vars.

### PR DESCRIPTION
**Pull request by siliconmeadow**
OpenID Connect credentials and some config to be provided via .env vars.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,4 +36,4 @@ certs/
 /private
 
 # Ignore config which may contain secrets
-**/openid_connect.client.azure_b2c.yml
```
Comment (MatthewWilkes) [inline .gitignore:39]: I think we should keep all openid_connect secrets in ignore. Maybe expand this to `**/openid_connect.client.*.yml`? That way, people still can't do an export and accidentally commit

File: assets/composer/settings.lando.php [hunk 2]
```diff
@@ -0,0 +1,169 @@
+<?php
```
Comment (MatthewWilkes) [inline assets/composer/settings.lando.php:1]: Do we support Lando?

File: assets/composer/settings.local.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+if (getenv('OPENID_CONNECT_PARAMS') == true) {
+  $openid_config_params = json_decode(getenv('OPENID_CONNECT_PARAMS'), true);
```
Comment (MatthewWilkes) [inline assets/composer/settings.local.php:4]: I don't think we need to manage any of these in the environment variables, but doesn't do any harm to have them.

File: assets/composer/settings.local.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  $config['openid_connect.settings']['always_save_userinfo'] = $openid_config_params['always_save_userinfo'];
+  $config['openid_connect.settings']['connect_existing_users'] = $openid_config_params['connect_existing_users'];
+  $config['openid_connect.settings']['override_registration_settings'] = $openid_config_params['override_registration_settings'];
+  $config['openid_connect.settings']['end_session_enabled'] = $openid_config_params['end_session_enabled'];
+  $config['openid_connect.settings']['user_login_display'] = $openid_config_params['user_login_display'];
+  foreach($openid_config_params['clients'] as $client => $value) {
+    $config['openid_connect.client.' . $client]['status'] = $value['status'];
```
Comment (MatthewWilkes) [inline assets/composer/settings.local.php:11]: Can we just set the whole client in from the environment variable, not try and put some in the config and other bits in the environment?  We don't have flow or profiles in this, which could cause issues, and we are forcing two configs to be present, but we don't necessarily want those two in each site.

File: config/default/openid_connect.settings.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+always_save_userinfo: true
```
Comment (MatthewWilkes) [inline config/default/openid_connect.settings.yml:1]: I like this being in the checked in config

**Review by MatthewWilkes — CHANGES_REQUESTED** [review]

**Review by MatthewWilkes — APPROVED** [review]
Provisionally approving pending 2nd PR addressing comments

File: assets/composer/settings.lando.php (hunk 2 above)
Comment (siliconmeadow) [reply assets/composer/settings.lando.php:1]: We don't, but LGD does. That lando config stuff is there in intra's `settings.local.php`.

I'll take it out on both sites when I do the security updates.

File: assets/composer/settings.local.php (hunk 3 above)
Comment (siliconmeadow) [reply assets/composer/settings.local.php:4]: That's a hybrid blob of data for for the openid_connect module and x number of clients. One var per env. And the reason I've stored the openid_connect module's variables - there is one which we are certain to vary between envs: the placement of the buttons - above, or replacing the login form.

File: assets/composer/settings.local.php (hunk 4 above)
Comment (siliconmeadow) [reply assets/composer/settings.local.php:11]: I did put some effort into do that and removing the config, but so far it's a no go.

**Pull request merged**